### PR TITLE
feat(app): Consolidate AWS / Azure Export Examples

### PR DESCRIPTION
**Pull request by AlexCuse**
Remove AWS / Azure export examples and replace with profiles that
include example formatted values for simple-filter-xml-mqtt

Signed-off-by: Alex Ullrich <alex.ullrich@gmail.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/Contributing.md.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: application-services/custom/cloud-export-mqtt/main.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	xml, ok := data.(string)
+	if !ok {
+		return false, errors.New("printXMLToConsole: Data received is not the expected 'string' type")
+
+	}
+
+	lc.Debug(xml)
+	ctx.SetResponseData([]byte(xml))
+	return true, xml
+}
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/main.go:111]: Remove

File: application-services/custom/cloud-export-mqtt/pkg/transforms/conversion.go [hunk 2]
```diff
@@ -18,19 +17,25 @@ func NewConversion() Conversion {
...
-	if event, ok := params[0].(models.Event); ok {
+	if event, ok := data.(models.Event); ok {
 		readings := map[string]interface{}{}
 
 		for _, reading := range event.Readings {
-			readings[reading.Name] = reading.Value
+			br, ok := reading.(models.SimpleReading)
+
+			if !ok {
+				continue
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/pkg/transforms/conversion.go:35]: Add log message or command why?

File: application-services/custom/cloud-export-mqtt/res/aws/configuration.toml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    path = "redisdb"
+      [Writable.InsecureSecrets.DB.Secrets]
+      username = ""
+      password = ""
+    [Writable.InsecureSecrets.mqtt]
+    path = "mqtt"
+      [Writable.InsecureSecrets.mqtt.Secrets]
+      username = "user-name"
+      password = "p@ssword"
+[Service]
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/aws/configuration.toml:18]: Add blank line above

File: application-services/custom/cloud-export-mqtt/res/aws/configuration.toml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+HealthCheckInterval = "10s"
+Host = "localhost"
+Port = 59780 # Adjust if running multiple examples at the same time to avoid duplicate port conflicts
+ServerBindAddr = "" # if blank, uses default Go behavior https://golang.org/pkg/net/#Listen
+StartupMsg = "This is a sample Filter/XML/MQTT Transform Application Service"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/aws/configuration.toml:23]:
```suggestion
StartupMsg = "This is a sample Cloud Export to AWS"

File: application-services/custom/cloud-export-mqtt/res/aws/configuration.toml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+# SecretStore is required when Store and Forward is enabled and running with security
+# so Database credentials can be pulled from Vault.
+# Note when running in docker from compose file set the following environment variables:
+#   - SecretStore_Host: edgex-vault
+[SecretStore]
+Type = "vault"
+Host = "localhost"
+Port = 8200
+Path = "app-simple-filter-xml-mqtt/"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/aws/configuration.toml:48]:
```suggestion
Path = "app-cloud-export-mqtt/"
```

File: application-services/custom/cloud-export-mqtt/res/aws/configuration.toml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+Protocol = "http"
+TokenFile = "/tmp/edgex/secrets/app-simple-filter-xml-mqtt/secrets-token.json"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/aws/configuration.toml:50]:
```suggestion
TokenFile = "/tmp/edgex/secrets/app-cloud-export-mqtt/secrets-token.json"
```

File: application-services/custom/cloud-export-mqtt/res/azure/configuration.toml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    path = "redisdb"
+      [Writable.InsecureSecrets.DB.Secrets]
+      username = ""
+      password = ""
+    [Writable.InsecureSecrets.mqtt]
+    path = "mqtt"
+      [Writable.InsecureSecrets.mqtt.Secrets]
+      username = "user-name"
+      password = "p@ssword"
+[Service]
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/azure/configuration.toml:18]: all blank line above

File: application-services/custom/cloud-export-mqtt/res/azure/configuration.toml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+# SecretStore is required when Store and Forward is enabled and running with security
+# so Database credentials can be pulled from Vault.
+# Note when running in docker from compose file set the following environment variables:
+#   - SecretStore_Host: edgex-vault
+[SecretStore]
+Type = "vault"
+Host = "localhost"
+Port = 8200
+Path = "app-simple-filter-xml-mqtt/"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/azure/configuration.toml:48]:
```suggestion
Path = "app-cloud-export-mqtt/"
```

File: application-services/custom/cloud-export-mqtt/res/azure/configuration.toml [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+Protocol = "http"
+TokenFile = "/tmp/edgex/secrets/app-simple-filter-xml-mqtt/secrets-token.json"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/azure/configuration.toml:50]:
```suggestion
TokenFile = "/tmp/edgex/secrets/app-cloud-export-mqtt/secrets-token.json"
```

File: application-services/custom/cloud-export-mqtt/res/azure/configuration.toml [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+# Example of App Service specific structured custom configuration.
+# This map into the transforms.MQTTSecretConfig structure
+# See https://docs.edgexfoundry.org/2.0/microservices/application/AdvancedTopics/#custom-configuration
+[MqttExportConfig]
+BrokerAddress = "tcp://my-iot-hub.azure-devices.net:1883"
+ClientId = "my-iot-device"
+AutoReconnect = true
+Topic = "devices/my-iot-device/messages/events/" # iot device name MUST match client ID
+AuthMode = "none"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/azure/configuration.toml:95]:
```suggestion
AuthMode = "none" # Change if auth required. See https://docs.edgexfoundry.org/2.0/microservices/application/BuiltIn/#mqtt-export for details
```

File: application-services/custom/cloud-export-mqtt/res/aws/configuration.toml [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+# Example of App Service specific structured custom configuration.
+# This map into the transforms.MQTTSecretConfig structure
+# See https://docs.edgexfoundry.org/2.0/microservices/application/AdvancedTopics/#custom-configuration
+[MqttExportConfig]
+BrokerAddress = "tcps://***.iot.us-west-2.amazonaws.com:8883"
+ClientId = "testThing"
+AutoReconnect = true
+Topic = "$aws/things/testThing/shadow/update" # iot device name MUST match client ID
+AuthMode = "none"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/aws/configuration.toml:95]:
```suggestion
AuthMode = "none"  # Change if auth required. See https://docs.edgexfoundry.org/2.0/microservices/application/BuiltIn/#mqtt-export for details
```

File: application-services/custom/cloud-export-mqtt/res/azure/configuration.toml [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    path = "mqtt"
+      [Writable.InsecureSecrets.mqtt.Secrets]
+      username = "user-name"
+      password = "p@ssword"
+[Service]
+HealthCheckInterval = "10s"
+Host = "localhost"
+Port = 59780 # Adjust if running multiple examples at the same time to avoid duplicate port conflicts
+ServerBindAddr = "" # if blank, uses default Go behavior https://golang.org/pkg/net/#Listen
+StartupMsg = "This is a sample Filter/XML/MQTT Transform Application Service"
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/azure/configuration.toml:23]:
```suggestion
StartupMsg = "This is a sample Cloud Export to AZURE"
```

File: application-services/custom/cloud-export-mqtt/res/configuration.toml [hunk 13]
```diff
@@ -0,0 +1,91 @@
+[Writable]
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/res/configuration.toml:1]: Default config can be removed. 
Need to add a README explaining how to run example using one of the profiles. i.e.
```
./app-service -p aws
of
./app-service -p azure
```

and maybe have something about the TransformToCloudFormat function is a simple example and real-world will need more use case specific??

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: application-services/custom/cloud-export-mqtt/main.go [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+	if err := config.Validate(); err != nil {
+		lc.Errorf("Custom Config failed validation: %s", err.Error())
+		os.Exit(-1)
+	}
+
+	// 4) This is our pipeline configuration, the collection of functions to
+	// execute every time an event is triggered.
+	if err := service.SetFunctionsPipeline(
+		transforms.NewFilterFor(deviceNames).FilterByDeviceName,
+		transforms.NewConversion().TransformToXML,
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply application-services/custom/cloud-export-mqtt/main.go:73]: Need to create the `cloudTransforms` alias ...

File: application-services/custom/cloud-export-mqtt/pkg/transforms/conversion.go (hunk 2 above)
Comment (AlexCuse) [reply application-services/custom/cloud-export-mqtt/pkg/transforms/conversion.go:35]: I guess really theres nothing stopping us from just throwing binary data in there :)

File: application-services/custom/cloud-export-mqtt/main.go (hunk 14 above)
[the remark this answers is not in the document]
Comment (AlexCuse) [reply application-services/custom/cloud-export-mqtt/main.go:73]: I removed the filter too

File: application-services/custom/cloud-export-mqtt/README.md [hunk 15]
```diff
@@ -0,0 +1,11 @@
+# cloud export mqtt
+
+This example service is meant to demonstrate using a custom transformation function (example only) along with the MQTTSecretSender to deliver edgex data to a cloud provider over MQTT.  This is done via profiles that really only differ in the driver settings.
```
Comment (lenny-intel) [inline application-services/custom/cloud-export-mqtt/README.md:3]:
```suggestion
This example service is meant to demonstrate using a custom transformation function (example only) along with the MQTTSecretSender to deliver edgex data to a cloud provider over MQTT.  This is done via profiles that really only differ in the connect settings.
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by AlexCuse — COMMENTED** [review]
fix(app): PR Review

This should be something you can squash :)

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**